pkg/metrics/metrics.jsx

```diff
@@ -431,7 +432,9 @@ class CurrentMetrics extends React.Component {
 
         data.forEach(temperatureSamples => decompress_samples(temperatureSamples, this.temperatureSamples));
 
-        this.cpuTemperature = parseInt(Math.max(...this.temperatureSamples[0]));
+        this.cpuTemperature = this.temperatureSamples[0].length > 0
+            ? Math.round(Math.max(...this.temperatureSamples[0]))
+            : NaN;
```

Comment:
IS the `parseInit` totally not needed anymore? Are the samples always send as integers not strings?

Reply:
After `JSON.parse` and decompressing samples are numbers are not strings. After looking at the code again I think I may have used `parseInt` as an improper way to round floats to integers.

Reply:
Ahh, that explains it.

Reply:
Assigning something to `NaN` still gives me a weird feeling. If I read the code we should fall into the `else` statement which doesn't seem to be something we want?

We should just hold off displaying / doing anything until we have valid data.

Reply:
Note that we check `!isNan(this.cpuTemperature` in the render function.